cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp

```diff
@@ -0,0 +1,1230 @@
+                                              List)) {
+                    lsum_ptr[id] = mrgsum;
+                }
+                if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+                    lsum2_ptr[id] = mrgsum2;
+                }
+                if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+                    lrc_ptr[id] += rcnt;
+                }
+                if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+                    lsum2cent_ptr[id] = mrgsum2cent;
+                }
+                if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+                    lmean_ptr[id] = mrgmean;
+                }
+            }
+        }
+    }
+
+    if (0 == id) {
+        if constexpr (check_mask_flag(bs_list::min, List)) {
+            rmin_ptr[group_id] = mrgmin;
+        }
+        if constexpr (check_mask_flag(bs_list::max, List)) {
+            rmax_ptr[group_id] = mrgmax;
+        }
+        if constexpr (check_mask_flag(bs_list::sum | bs_list::mean | sum2cent_based_stat, List)) {
+            rsum_ptr[group_id] = mrgsum;
+        }
+        if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+            rsum2_ptr[group_id] = mrgsum2;
+        }
+        if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+            rsum2cent_ptr[group_id] = mrgsum2cent;
+        }
+
+        if constexpr (!DefferedFin) {
+            Float mrgvariance = mrgsum2cent / (mrgvectors - Float(1));
+            Float mrgstdev = (Float)sqrt(mrgvariance);
+            if constexpr (check_mask_flag(bs_list::mean, List)) {
+                rmean_ptr[group_id] = mrgmean;
+            }
+            if constexpr (check_mask_flag(bs_list::sorm, List)) {
+                rsorm_ptr[group_id] = mrgsum2 / mrgvectors;
+            }
+            if constexpr (check_mask_flag(bs_list::varc, List)) {
+                rvarc_ptr[group_id] = mrgvariance;
+            }
+            if constexpr (check_mask_flag(bs_list::stdev, List)) {
+                rstdev_ptr[group_id] = mrgstdev;
+            }
+            if constexpr (check_mask_flag(bs_list::vart, List)) {
+                rvart_ptr[group_id] = mrgstdev / mrgmean;
+            }
+        }
+    }
+}
+
+template <typename Float, bs_list List>
+std::tuple<ndresult<Float, List>, sycl::event> compute_kernel_dense_impl<Float, List>::merge_blocks(
+    ndbuffer<Float, List>&& ndbuf,
+    std::int64_t column_count,
+    std::int64_t block_count,
+    const bk::event_vector& deps) {
+    ONEDAL_PROFILER_TASK(merge_blocks, q_);
+
+    ONEDAL_ASSERT(column_count > 0);
+    ONEDAL_ASSERT(block_count > 0);
+
+    const bool distr_mode = comm_.get_rank_count() > 1;
+    auto ndres = ndresult<Float, List>::empty(q_, column_count, distr_mode);
+
+    // ndres asserts
+    ASSERT_IF(bs_list::min, ndres.get_min().get_count() == column_count);
```

Comment:
What is `ASSERT_IF`? Can you please use `ONEDAL_ASSERT` instead?

Reply:
Please find this define at the file's beginning, it was aimed to make assertion only in case if mentioned buffer is required and improve code's visibility by hiding lot of 'if constexpr(...) {'